### ISSUE TITLE
Installation failure with R 4.2.0

### ISSUE THREAD

**Issue by Deleetdk** [rstudio/httpuv#341]
I am on Mint 19.3 (Ubuntu 18.04 based) and just upgraded to R 4.2.0:

```
> sessionInfo()
R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Linux Mint 19.3

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1
```

I cannot install this package again. The CRAN and Github versions give the same error:

```
> devtools::install_github("rstudio/httpuv")
Downloading GitHub repo rstudio/httpuv@HEAD
✔  checking for file ‘/tmp/RtmpThQuzi/remotes74c81dee06d5/rstudio-httpuv-220319d/DESCRIPTION’ ...
─  preparing ‘httpuv’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘httpuv_1.6.5.9000.tar.gz’
   
Installing package into ‘/home/emil/R/x86_64-pc-linux-gnu-library/4.2’
(as ‘lib’ is unspecified)
* installing *source* package ‘httpuv’ ...
** using staged installation
** libs
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c RcppExports-legacy.cpp -o RcppExports-legacy.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c RcppExports.cpp -o RcppExports.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c callback.cpp -o callback.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c callbackqueue.cpp -o callbackqueue.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c filedatasource-unix.cpp -o filedatasource-unix.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c filedatasource-win.cpp -o filedatasource-win.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fs.cpp -o fs.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c gzipdatasource.cpp -o gzipdatasource.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c http.cpp -o http.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c httprequest.cpp -o httprequest.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c httpresponse.cpp -o httpresponse.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c httpuv.cpp -o httpuv.o
gcc -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c md5.c -o md5.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c mime.cpp -o mime.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c socket.cpp -o socket.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c staticpath.cpp -o staticpath.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c thread.cpp -o thread.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c timegm.cpp -o timegm.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c utils.cpp -o utils.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c uvutil.cpp -o uvutil.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c webapplication.cpp -o webapplication.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c websockets-base.cpp -o websockets-base.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c websockets-hixie76.cpp -o websockets-hixie76.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c websockets-hybi03.cpp -o websockets-hybi03.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c websockets-ietf.cpp -o websockets-ietf.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c websockets.cpp -o websockets.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG -Ilibuv/include -pthread -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include' -I'/home/emil/R/x86_64-pc-linux-gnu-library/4.2/later/include'   -fvisibility=hidden -DSTRICT_R_HEADERS -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c winutils.cpp -o winutils.o
cp -p -f libuv/m4/lt_obsolete.m4 libuv/m4/lt~obsolete.m4
cd libuv; \
if ! command -v automake >/dev/null 2>&1 ; then \
	echo "automake not found. Touching files so configure will not try to run automake."; \
	touch aclocal.m4; \
	touch -r aclocal.m4 configure Makefile.in; \
else \
	echo "automake found. Running autogen.sh."; \
	sh autogen.sh; \
fi; \
chmod +x configure; \
CC="gcc" CFLAGS="-g -O2 -fdebug-prefix-map=/build/r-base-kt0bjq/r-base-4.2.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -fpic -fvisibility=hidden" AR="ar" RANLIB="ranlib" LDFLAGS="-Wl,-Bsymbolic-functions -Wl,-z,relro" ./configure "--quiet"
automake found. Running autogen.sh.
+ libtoolize --copy
autogen.sh: 43: autogen.sh: libtoolize: not found
make --directory=libuv \
	HAVE_DTRACE=0
make[1]: Entering directory '/tmp/RtmpPeH3gh/R.INSTALLd6474822830/httpuv/src/libuv'
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash '/tmp/RtmpPeH3gh/R.INSTALLd6474822830/httpuv/src/libuv/missing' aclocal-1.16 -I m4
/tmp/RtmpPeH3gh/R.INSTALLd6474822830/httpuv/src/libuv/missing: line 81: aclocal-1.16: command not found
WARNING: 'aclocal-1.16' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <https://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/autoconf>
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
Makefile:1529: recipe for target 'aclocal.m4' failed
make[1]: *** [aclocal.m4] Error 127
make[1]: Leaving directory '/tmp/RtmpPeH3gh/R.INSTALLd6474822830/httpuv/src/libuv'
Makevars:87: recipe for target 'libuv/.libs/libuv.a' failed
make: *** [libuv/.libs/libuv.a] Error 2
ERROR: compilation failed for package ‘httpuv’
* removing ‘/home/emil/R/x86_64-pc-linux-gnu-library/4.2/httpuv’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpThQuzi/file74c83ade8825/httpuv_1.6.5.9000.tar.gz’ had non-zero exit status
```

Looking over the results on Google, I find it may be related to `autoconf` and `automake`. https://github.com/rstudio/httpuv/issues/280

Mine are both the newest versions and reinstalling them didn't do anything:

```
emil@desktop:~$ sudo apt-get install automake
[sudo] password for emil:             
Reading package lists... Done
Building dependency tree       
Reading state information... Done
automake is already the newest version (1:1.15.1-3ubuntu2).
The following packages were automatically installed and are no longer required:
  libllvm13:i386 libstd-rust-1.53 linux-headers-4.15.0-167
  linux-headers-4.15.0-167-generic linux-headers-5.4.0-99-generic
  linux-hwe-5.4-headers-5.4.0-99 vimix-gtk-theme
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 26 not upgraded.
emil@desktop:~$ apt install autoconf
Reading package lists... Done
Building dependency tree       
Reading state information... Done
autoconf is already the newest version (2.69-11).
The following packages were automatically installed and are no longer required:
  libllvm13:i386 libstd-rust-1.53 linux-headers-4.15.0-167
  linux-headers-4.15.0-167-generic linux-headers-5.4.0-99-generic
  linux-hwe-5.4-headers-5.4.0-99 vimix-gtk-theme
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 26 not upgraded.
```

Finally I figured out that the crucial error line given is `autogen.sh: 43: autogen.sh: libtoolize: not found`, which is mentioned here https://github.com/beakerbrowser/beaker/issues/54 as the missing system dependency of `libtool` (`apt install libtool`). After installing that, httpuv installed correctly.

**Comment by jcheng5** [comment]
Glad you were able to figure it out!

**Comment by jamesdalg** [comment]
This also solved my exact issue when updating R from 3.6.0 to 4.2.1. Thanks for contributing it!
